core/modules/entity/entity.module

```diff
@@ -140,6 +140,34 @@ function entity_view_mode_exists($machine_name, $element, $form_state) {
   return !empty($entity_info['view modes'][$machine_name]);
 }
 
+/**
+ * Determines whether the given user has access to an entity.
+ *
+ * @param string $op
+ *   The operation being performed. One of 'view', 'update' or 'delete'.
+ * @param string $entity_type
+ *   The entity type of the entity to check for.
+ * @param EntityInterface $entity
+ *   Optionally an entity to check access for. If no entity is given, it will be
+ *   determined whether access is allowed for all entities of the given type.
+ * @param User $account
+ *   The user to check for. Leave it to NULL to check for the global user.
+ *
+ * @return bool
+ *   Whether access is allowed or not. If the entity type does not specify any
+ *   access information, NULL is returned.
+ *
+ * @see entity_type_supports()
+ * @deprecated since 1.0.0
```

Comment:
It doesn't seem like this should include `@deprecated since 1.0.0`. Functional wrappers around OO methods are not discouraged, they're just a different option.